src/Kestrel.Core/Internal/Http/OutputProducer.cs

```diff
@@ -148,6 +149,41 @@ public void Abort(Exception error)
             return FlushAsync(writableBuffer, cancellationToken);
         }
 
+        public async Task WriteAsync(
+            Stream stream,
+            CancellationToken cancellationToken)
+        {
+
+            Monitor.Enter(_contextLock);
```

Comment:
Not sure about lock here :-/

Reply:
Yea, I'm not sure about this either. async + Monitor... is a bad combination